### ISSUE TITLE
Version 1.7 in PROD - Instructions and Bug Reporting

### ISSUE THREAD

**Issue by jamesmrollins** [lzim/teampsd#287]
We are excited to move Version 1.7 (SP) to PROD.  Please keep some things in mind:

1.  While don't anticipate issues with previously saved experimental files, there is always that risk.  Especially with respect to our work with time stepping all the models to zero.  @lzim said to me that exports were run in case any one file is corrupt.  Therefore, we should use previous runs carefully, using export files to gather text and values to rerun if we get wonky results.

2.  I have mapped "100a1_abc_teama_2018_11_01.xlsx to the MTL" Facilitate Pilot world.  

3.  If something comes up that you need me to address, I will be available through the holiday.  The DEV team will be working as well.  Please don't hesitate to report something to me.

4.  The "i" information has not been loaded yet.  I will attempt to get that done today in between cooking and prep for TURKEY DAY!  

Cheers! James

**Comment by TomRust** [comment]
Awesome James!  

One bug found: We have implementing MBC actually **increasing** the time low symptom patients are in care, but the model diagram in the UI has it as decreasing.  I've included a screenshot, with the arrow polarity problem circled.  It should be changed from a "-" to a "+" in all three settings.  

Thanks!

![image](https://user-images.githubusercontent.com/32961420/48856470-75a4ca00-ed84-11e8-9299-d03c52029f70.png)

**Comment by TomRust** [comment]
Can we get nine (9) more variables in the drop down list on the Results Dashboard in SP?

GMH to SMH Wait Time for Step up -- swap in as GMH default chart #5
GMH to PC/PCMHI Wait Time to Step down -- swap in as GMH default chart #6
PC/PCMHI to GMH Wait Time to Step up -- swap in as PC default chart #5
PC/PCMHI to SMH Wait Time to Step up -- swap in as PC default chart #6
SMH to GMH Wait Time to Step down -- swap in as SMH default chart #5
SMH to PC/PCMHI Wait Time to Step down -- swap in as SMH default chart #6
PC/PCMHI New Patient Wait Time to Start
GMH New Patient Wait Time to Start
SMH New Patient Wait Time to Start

THANKS!!!!!

**Comment by dlkibbe** [comment]
@staceypark @jamesmrollins - sorry for my bad memory/lack of practice about groups and mapping myself to a world so I can run simulations using 1.7 PROD. Stacey, I thought you had an Epicenter instructions document in the Resources file on GitHub. Can you email me the instructions for mapping myself?

**Comment by jamesmrollins** [comment]
@dlkibbe I see that you are already mapped to every one of the MTL Demo groups.  Are you not able to get in, or are you trying to teach it?

**Comment by jamesmrollins** [comment]
@TomRust  Reported the two issues above.

**Comment by dlkibbe** [comment]
Good morning @jamesmrollins- working in CC PROD model. Trying to simulate increased referral rate. The maximum number on the slider is 30. The referral rate base case is 31.91. See screen shot below.
![image](https://user-images.githubusercontent.com/31182258/48954072-01019500-ef16-11e8-8f9b-1f34a3fa03c9.png)

**Comment by jamesmrollins** [comment]
Hi @dkibbe, We have a few instances of this, including RVI (I think). @lzim, @trust, has anyone reviewed the ranges of sliders for all models? I need recommendations, still. Thanks, James

Get Outlook for iOS<https://aka.ms/o0ukef>

________________________________

**Comment by lzim** [comment]
@TomRust have you had any time to review the ranges in relationship to the test dataset we are using?

It's true that I already reported this CC example to @jamesmrollins and I think I pinged you, because I wasn't sure if this was both a model and DEV issue.

Thanks!

**Comment by TomRust** [comment]
I already responded directly to James, but can now post here, too:
I think 52 weeks for RVI max is extreme, but I'm usually off by an order of magnitude between my initial guess at a max value and what we eventually end up using.  :)  If any team wants to see their patients an average of once per year...I guess we should let them.  No patients are harmed, right?

As to Referral Rate max: Anything less than 1/4th of the max appointments supply is reasonable.  If we think the average patient only gets 1 appointment, then it would be reasonable for the supply and referral rates to have the same max.  I picked 1/4th because I can't imagine a team where the average patient gets less than 4 visits total in their entire episode of care (sure, some patients could get that few, but the average patient? when some patients stay in care for years?).  

I'm just guessing though!

**Comment by jamesmrollins** [comment]
@lzim and @TomRust I am documenting a text conversation I had with @lzim regarding Tom's recommendations above.  She concurs that with an RVI as high as 52 weeks, and Tom's recommendation that 1/4 th or less of the Appointment Supply should be the Referral Rate maximum, so 200 x .25 = 50.  I also recommended we use the same maximums in the MM model.  @lzim concurred.  I will refer these changes to DEV tonight for update by tomorrow.

Thanks,
James

**Comment by jamesmrollins** [comment]
@lzim @TomRust @dlkibbe @branscombj @staceypark @dlounsbu
MTL Ver 1.7 is now live. . . 

Enjoy!

**Comment by jamesmrollins** [comment]
@lzim @TomRust @dlkibbe @branscombj @staceypark @dlounsbu
Darn! 
Just found three "i" pop ups that we missed.  I need text for:
"PC/PCMHI Community Care Rate"
"GMH Community Care Rate"
"SMH Community Care Rate"

After I get the text, I should be able to quickly map these by tomorrow morning.

Thanks,
James

**Comment by jamesmrollins** [comment]
@TomRust and @lzim  We devised the following, which will be sent to dev for tonight's push, unless otherwise amended:

"XXX The number of XXX patients per week that start in Community Care."

Thanks,
James

**Comment by dlkibbe** [comment]
@jamesmrollins Two things about this text in red: "Filename entered into the filed above must match filename exactly."
 1) can you please correct the spelling of "into the filed" to "into the field"; 2) can you please change the "filename" to "file name" as it appears in the box as 2 words.
![image](https://user-images.githubusercontent.com/31182258/49121668-4fe55c80-f27f-11e8-9994-45c06ce93dbf.png)

**Comment by dlkibbe** [comment]
@jamesmrollins - please edit the text in the pop-up box when you click to save a run. It currently says:
"Do you want to continue saving, without complete entries in the Expanded Outputs Tile text fields ?"

It should say: "Do you want to continue saving without complete entries in the Text fields?"

![image](https://user-images.githubusercontent.com/31182258/49122148-3a713200-f281-11e8-8fa1-8250847d69c2.png)

**Comment by dlounsbu** [comment]
In the 1.7 Prod, we noted that, on the VA Studio computer in St. Louis, the bar chart graph windows in PSY graphs for RVI and for engagement time were truncated, so that when Lindsey displayed the graph, the window did not fully display longitudinally. Only part of the third set of bars were shown. The chart window could not be resized. However, now, on my laptop, there is no issue!!

**Comment by dlkibbe** [comment]
@jamesmrollins - can you please remove the hyphen and the (service) in parentheses from this i box?

![image](https://user-images.githubusercontent.com/31182258/49122632-40681280-f283-11e8-8dc6-e087a6765da0.png)

**Comment by dlkibbe** [comment]
@jamesmrollins - Agg Model, True Missed Appointment % "i" box under the Experiments section 
The True Missed Appointment % "i" box has no title or text.

![image](https://user-images.githubusercontent.com/31182258/49122729-a359a980-f283-11e8-8a5e-2150d5a28a58.png)

**Comment by dlounsbu** [comment]
@jamesmrollins :  here is where we updated the green and red service proportions: https://github.com/lzim/teampsd/blob/master/resources/master_crosswalk/info/AGGModelInfo.md

**Comment by lzim** [comment]
@jamesmrollins one more :smile:

We need the arrow from "Review Rate" to "Treatment Decision Rate" to be available in default view in Agg, right now it is revealed with a complexity reveal instead.

See the screen shots.

![arrow_review_rate2](https://user-images.githubusercontent.com/7454688/49123253-93d76200-f27d-11e8-9a41-dac5eec9e5f5.png)
![arrow_review_rate](https://user-images.githubusercontent.com/7454688/49123254-946ff880-f27d-11e8-8202-62431581fa48.png)

**Comment by dlkibbe** [comment]
@jamesmrollins - Agg, Team Data Table "i" information
There are "X's"  in the middle row under the columns titled Table Variable  and Data and Estimation Description

![image](https://user-images.githubusercontent.com/31182258/49124707-685b7400-f28b-11e8-8a0a-e3a035d628b9.png)

**Comment by jamesmrollins** [comment]
@lzim @TomRust @dlkibbe @dlounsbu @branscombj 

I scrubbed all the model "i" information popup to find any other un-linked information.  The problem did not appear in  CC, MM or PSY.  But was prolific in AGG and SP.  Should be fixed by morning.  I am not sure what happened, but we are going to investigate and get to the bottom of it.  I have also reported the edits for the save dialogue box, the reveal arrow issue, and the file name in the Team Data Menu Maintenance. 

@dlounsbu  regarding the PSY Chart rendering issue.  Since the html dynamically adjusts the screen based on the screen resolution, it is difficult to design to that many possible variations. However, there are a few things to try when grappling with resolution issues.  If in Chrome, there is a way to adjust your screen zoom in the upper right (see attached screen capture).  Also, you can drag the bar charts around the screen, if too many things are open (such as the Text Window), it may be adjusting size to accommodate all the objects on the screen. 

![image](https://user-images.githubusercontent.com/30132017/49133822-5501d600-f296-11e8-8e73-786f3feeebdc.png)

Finally, if the team needs to edit model-popup-information, it can be done directly and in real-time by downloading the model-popup-information.xlsx file from the Epicenter Interface Directory - Dynamic Data file.  Make your wording adjustment, delete the file currently in the Interface Directory, and then upload the edited file.  That will reduce the number hand-offs for updating and reduce the potential for error. 

![image](https://user-images.githubusercontent.com/30132017/49134262-d60d9d00-f297-11e8-81ce-d22b52ccda98.png)

![image](https://user-images.githubusercontent.com/30132017/49134341-1bca6580-f298-11e8-8d7f-d119aa527890.png)


Good Night,
James

**Comment by jamesmrollins** [comment]
@TomRust @lzim @dlkibbe @branscombj @dlounsbu 
I traced the problems for the "i" information back to there were simply missing from the model-popup-information file.  In some cases they are missing from the R code markup.  I verified these variables are present in the master_crosswalk_table, but were missed when the crosswalk was copied for "i" information edit, and again when they were merged.  In any case, I don't feel that managing lists with spreadsheets is a good long-term solution and the process between R code markup and spreadsheets is not really clear nor easy.

In any case, we still need "i" explanations for the variables below:

AGG
"Patient Start Rate"

SP
GMH High Risk Flag To Inpatient or Residential Rate
PC/PCMHI High Risk Flag To Inpatient or Residential Rate
SMH High Risk Flag To Inpatient or Residential Rate
User GMH to SMH Engagement Time Before Step up
User GMH to PC/PCMHI Engagement Time Before Step down
User PC/PCMHI to GMH Engagement Time Before Step up
User PC/PCMHI to SMH Engagement Time Before Step up
User SMH to PC/PCMHI Engagement Time Before Step down
User SMH to GMH Engagement Time Before Step down

**Comment by dlkibbe** [comment]
@jamesmrollins while reviewing the SP module this morning on @lzim MAC using Safari, there were blanks in the i information on the model diagram. The i information boxes on my Dell were populated using Chrome. Not sure why this would happen.

**Comment by jamesmrollins** [comment]
@dlkibbe  Normally issues such as the one you describe have to availability of the server to feed the information to your browser.  Server availability can be affected by everything from internet traffic, to the number of calls being made on the server at the same time.  This is normally rectified by going back to home and reloading the page (session).  If the issue persists, please go to "inspect" mode and send me the feedback from the console.  Thanks, James

**Comment by TomRust** [comment]
Hi @jamesmrollins !

We found a bug in CC: the widgets are all working perfectly (ie, pulling in the values at time=0,not time=-500), but the BC button for Referral Rate is still pulling in the value at Time=-500.  
It is the only BC button in CC.

See screenshot.

![image](https://user-images.githubusercontent.com/32961420/49250267-6bba4100-f3ec-11e8-8f04-7864337354d1.png)

**Comment by jamesmrollins** [comment]
@TomRust  OK, I'll check it out.

**Comment by TomRust** [comment]
100a1_abc_teama_2018_11_01

**Comment by jamesmrollins** [comment]
@TomRust Great, I see the problem.  Thanks.  I will run through the rest of the BCs again, to make sure they don't have any errors.

**Comment by TomRust** [comment]
Hi @jamesrollins !

Whoops!  Well…we don’t need those this week, as we’ve already wrapped up shooting those modules.  :)

Are you still going to create a combined iText table in GitHub?  Or, should I put the text for these variables into the current, model-specific tables?

--TOM


@TomRust<https://github.com/TomRust> @lzim<https://github.com/lzim> @dlkibbe<https://github.com/dlkibbe> @branscombj<https://github.com/branscombj> @dlounsbu<https://github.com/dlounsbu>
I traced the problems for the "i" information back to there were simply missing from the model-popup-information file. In some cases they are missing from the R code markup. I verified these variables are present in the master_crosswalk_table, but were missed when the crosswalk was copied for "i" information edit, and again when they were merged. In any case, I don't feel that managing lists with spreadsheets is a good long-term solution and the process between R code markup and spreadsheets is not really clear nor easy.

In any case, we still need "i" explanations for the variables below:

AGG
"Patient Start Rate"

SP
GMH High Risk Flag To Inpatient or Residential Rate
PC/PCMHI High Risk Flag To Inpatient or Residential Rate
SMH High Risk Flag To Inpatient or Residential Rate
User GMH to SMH Engagement Time Before Step up
User GMH to PC/PCMHI Engagement Time Before Step down
User PC/PCMHI to GMH Engagement Time Before Step up
User PC/PCMHI to SMH Engagement Time Before Step up
User SMH to PC/PCMHI Engagement Time Before Step down
User SMH to GMH Engagement Time Before Step down

**Comment by jamesmrollins** [comment]
@TomRust I think you should download a current copy from EPICENTER, update that table there and reload it back into EPICENTER.  Then send it to me, and I will continue my efforts to get it into GutHub.  I believe you sent me an R code converter.  Royal pain . . .

**Comment by TomRust** [comment]
Hi @jamesmrollins !  

One more tiny thing in CC:  The units on the chart of "Waiting to Start" should be "pts" not "appts"

![image](https://user-images.githubusercontent.com/32961420/49255386-dcb42580-f3f9-11e8-8d55-11dfa3468935.png)

**Comment by TomRust** [comment]
@jamesmrollins : Cool.  I'll download the "model-popup-information.xlsx" file, make the edits, and email it to you on Monday.

**Comment by jamesmrollins** [comment]
@TomRust Thanks, I'll report chart units of measure problem.  I found a similar BC problem in MM.  AGG and SP seem to be tracking ok.

**Comment by dlounsbu** [comment]
On phone 1.7, as there is no main model diagram, radio buttons from the main model diagram are not available to change the default output charts. This is an issue for changing services in AGG and MM for toggling has between different serivices.

**Comment by lzim** [comment]
Thanks @dlounsbu !

@jamesmrollins The issue is we were running sims from our phone and realized we cannot change the 6 default output charts to diagnostic cohorts (in MM) or services (in AGG), which we would do from a desktop using the main model diagram.

Any ideas?

**Comment by dlounsbu** [comment]
In MM, changing cohorts with radio buttons resets experiment sliders to bc! So, then you have to retype them. @jamesmrollins @TomRust @lzim

**Comment by jamesmrollins** [comment]
@dlounsbu  Here are some answers (?) to your observations above:

The phone view is really just the control panel from the expanded outputs screen, the charts, the experiments tile and the output charts all stacked one on top of the other.  We made a joint decision not to attempt to squeeze the main UI onto tiny phone screens.  So you are correct, there are limited views.  However, you can choose the Patient Cohorts mode in the Experiment Timeline to see the starting values for each cohort in the charts, as the model is run to equilibrium.  If the number is difficult to see as a graphic, try switching to table view. From the charts section, you should be able to see any variables in the sim UI for that setting.  

Regarding the radio button resetting the experiment sliders, you are correct there too.  When you select between pages, the experiment sliders will change.  That is because the experiment sliders bind their values with the page that is showing.  So when the new page reloads, the base case values start over, because it is a different page.  Since the experimental values have no affect on output until the model is run, I recommend adjusting the lesson so that adjusting experiment sliders is done just before selecting "Run" from the Experiment Timeline panel.

@lzim you cannot really change the default chart settings - they are hard coded consistent with the master_crosswalk_table.  Although while in a session you can change which charts you are viewing, when you change pages, or logout and back in, the hard-coded default charts are presented.  Unless I am missing something, the same functionality is presented in the tablet or phone view (at least on my iPhone SE).

**Comment by TomRust** [comment]
Hi @jamesmrollins ,

I don't see a variable "Patient Start Rate" in the master crosswalk for AGG.  We have text for "New Patient Start Rate," though, and "Starting a Service Rate."  Could we have accidentally dropped the "New" from the name during the shuffle?

Here is the info for SP: 
GMH   High Risk Flag To Inpatient or Residential Rate | The number of patients in General   mental healthcare with a high-risk flag who step up to either Inpatient or   Residential treatment each week.
-- | --
PC/PCMHI High   Risk Flag To Inpatient or Residential Rate | The number of   patients receiving mental healthcare in Primary Care with a high-risk flag   who step up to either Inpatient or Residential treatment each week.
SMH High Risk   Flag To Inpatient or Residential Rate | The number of patients in Specialty mental healthcare with a   high-risk flag who step up to either Inpatient or Residential treatment each   week.
User GMH to   SMH Engagement Time Before Step up | Set the   average length of time patients are engaged in General mental healthcare   before being recommended to step up to receive Specialty mental health   services. This value replaces the base case value estimated from team data.
User GMH to   PC/PCMHI Engagement Time Before Step down | Set the   average length of time patients are engaged in General mental healthcare   before being recommended to step down to receive mental health services in   the Primary Care setting. This value replaces the base case value estimated   from team data.
User PC/PCMHI   to GMH Engagement Time Before Step up | Set the average length of time patients are engaged in   receiving mental healthcare in the Primary Care setting before being   recommended to step up to receive General mental health services. This value   replaces the base case value estimated from team data.
User PC/PCMHI   to SMH Engagement Time Before Step up | Set the average length of time patients are engaged in   receiving mental healthcare in the Primary Care setting before being   recommended to step up to receive Specialty mental health services. This   value replaces the base case value estimated from team data.
User SMH to   PC/PCMHI Engagement Time Before Step down | Set the   average length of time patients are engaged in Specialty mental healthcare   before being recommended to step down to receive mental health services in   the Primary Care setting. This value replaces the base case value estimated   from team data.
User SMH to   GMH Engagement Time Before Step down | Set the   average length of time patients are engaged in Specialty mental healthcare   before being recommended to step down to receive General mental health care   services. This value replaces the base case value estimated from team data.

**Comment by jamesmrollins** [comment]
Hi @TomRust, thanks for knocking out those "i" information blurbs.  I have updated the model-popup-information.xlsx file in Epicenter. Regarding the AGG "Patient Start Rate."  I believe this is the variable shown below in the diagram in the SIM UI and in your model.  I copied the text from your model and put it into the model-popup file.  If you want an edit, just forward me the text, or go into the model-popup file and update from there. 

![image](https://user-images.githubusercontent.com/30132017/49750139-06254a80-fc5f-11e8-8f79-9f6d77dfbdec.png)

![image](https://user-images.githubusercontent.com/30132017/49750153-11787600-fc5f-11e8-8ba9-a688bd108587.png)

![image](https://user-images.githubusercontent.com/30132017/49750176-20f7bf00-fc5f-11e8-9596-aad1fe0e8163.png)

Thanks,
James
